### DISCUSSION TITLE
Content exclusion for JetBrains IDE

### DISCUSSION THREAD

**Discussion by whoisbarsuk** [community/community#77346] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi! I use JetBrains Copilot Plugin with my personal account subscription. I want to ensure Copilot Plugin scans only files inside IDE projects for suggestions and nothing except them.  I didn't find any proofs of that in the documentation.

**Comment by Akash1134** [comment]
Hi @ktobarsuk ,

Content Exclusion is currently only [supported](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-common-issues-with-github-copilot#github-copilot-content-exclusions-are-not-being-applied:~:text=Currently%20this%20feature%20is%20only%20supported%20in%20Visual%20Studio%20Code.) in VSCode IDE. 

GitHub Copilot content exclusions only apply to people who have been granted a seat as part of a **GitHub Copilot Business subscription** and are members of the same organization in which a content exclusion has been configured

Also, please note that **Content Exclusion feature is currently unavailable**. It will be reintroduced in the near future. For more information, see "[Copilot content exclusions – Temporary rollback and upcoming fix](https://github.blog/changelog/2023-11-20-copilot-content-exclusions-temporary-rollback-and-upcoming-fix/)" in the GitHub changelog.

Thanks!

**Comment by Akash1134** [comment, marked as answer]
## Copilot Content Exclusion Update

Following our [previous](https://github.blog/changelog/2023-11-08-copilot-content-exclusion-is-now-available-in-public-beta) communication dated November 8, 2023, regarding the temporary rollback of the Copilot content exclusions feature, we are pleased to announce the re-deployment of this feature with significant enhancements. The rollout will be progressive during the next 10 days as we monitor the behaviour.

With Content Exclusion, **GitHub Copilot Business customers will be able to prevent specified files or repositories from being used to inform code completion suggestions made by GitHub Copilot**. GitHub Copilot will not be available in excluded files. Organization administrators or repository owners can choose which files or repositories are excluded. [Learn more](https://docs.github.com/en/copilot/managing-copilot-business/configuring-content-exclusions-for-github-copilot).

### Overview of the Issue 👀 

Our team observed a critical issue where clients were incorrectly blocked from using Copilot due to the initial implementation of content exclusions. This was primarily caused by errors in fetching content exclusion policies from the client, leading to a temporary suspension of the feature.

### Actions Undertaken 🏁 

In response to this, our engineering team undertook a comprehensive review and rectification process. The issues identified in the client's code were addressed, and additional verifications were implemented on both server and client sides to prevent recurrence.

### New Enhancements in the Re-deployed Feature 🆕 

*   **Performance Update**: We have optimized the performance of the content exclusions feature, ensuring minimal impact on the user experience.
*   **Extended Coverage**: The feature now supports all our official Integrated Development Environments: Visual Studio, JetBrains IDEs, Visual Studio Code, and Vim/Neovim.

### Current Status 🟢 

*   Users with pre-existing content exclusion configurations will experience no change.
*   New and returning users can now utilize the enhanced feature across all supported IDEs.

**Up Next** ❓  We are closely monitoring the performance and user feedback post-deployment. The support for Copilot Chat is also in progress and will be part of the General Availability.
